client/signup/storageUtils.js

```diff
@@ -83,3 +83,25 @@ export const getSignupCompleteStepNameAndClear = () => {
 	clearSignupCompleteStepName();
 	return value;
 };
+export const setSignupStartTimeInSeconds = () =>
+	ignoreFatalsForSessionStorage( () =>
+		sessionStorage?.setItem( 'wpcom_signup_start_time_in_second', Math.floor( Date.now() / 1000 ) )
```

Comment:
I think we can use `performance.now()` which is more precise.

Reply:
Good idea. I've updated it into `performance.now()` instead. Other than the precision, it's immune to any potential system clock changes, so it's the better choice for measuring elapsed time by design.